G_channels.md

```diff
@@ -249,6 +249,24 @@ socket = assign(socket, :user, msg["user"])
 
 Sockets store assigned values as a map in `socket.assigns`.
 
+#### Failure Semantics
```

Comment:
@lancehalvorsen I'd love to find a new title for "Failure Semantics". How about, "Reliability Guarantees"? For me, that was the core question I wanted to see answered as a newcomer to Phoenix. I wanted to know what I could expect from the framework and what I'd have to build in order to get the guarantees I needed in my app.